### PR TITLE
Add ACES Filmic ToneMapping

### PR DESCRIPTION
**Pull request by FishOrBear**
ref: https://knarkowicz.wordpress.com/2016/01/06/aces-filmic-tone-mapping-curve/  

demo: https://qicad.com/ACESFilmToneMapping/examples/webgl_tonemapping.html

### CODE REVIEW

**Comment by mrdoob** [comment]
@bhouston are you okay with this addition?

File: examples/webgl_tonemapping.html [hunk 1]
```diff
@@ -73,7 +73,8 @@
 				Linear: THREE.LinearToneMapping,
 				Reinhard: THREE.ReinhardToneMapping,
 				Uncharted2: THREE.Uncharted2ToneMapping,
-				Cineon: THREE.CineonToneMapping
+				Cineon: THREE.CineonToneMapping,
+				ACESFilm: THREE.ACESFilmToneMapping
```
Comment (WestLangley) [inline examples/webgl_tonemapping.html]:
```javascript
ACESFilmic: THREE.ACESFilmicToneMapping
```
Please correct everywhere.

**Comment by WestLangley** [comment]
From the link:
>UPDATE: This is a very simple luminance only fit, which over saturates brights. This was actually something consistent with our art direction, but for a more realistic rendering you may want a more complex fit like this one from Stephen Hill.

That does appear to be true, but it is up to @bhouston.

File: examples/webgl_tonemapping.html (hunk 1 above)
Comment (FishOrBear) [reply examples/webgl_tonemapping.html]: Ok, I have modified the code.

**Comment by bhouston** [comment]
This seems decent.  I approve.  I haven't verified the math or compared to other aces output though but it seems plausible.

File: examples/webgl_tonemapping.html (hunk 1 above)
Comment (FishOrBear) [reply examples/webgl_tonemapping.html]: I didn't find the one I missed. Can you tell me?

**Comment by Mugen87** [comment]
@FishOrBear Can you please rebase in order to resolve the merge conflicts?

**Comment by bhouston** [comment]
BTW  I believe the correct name is ACESFilmicToneMapping not ACESFilmToneMapping.  Even Google search corrects you: https://www.google.ca/search?q=ACES+Film+Tone+Mapping

File: examples/webgl_tonemapping.html (hunk 1 above)
Comment (bhouston) [reply examples/webgl_tonemapping.html]: Ah, I missed this comment.

**Comment by WestLangley** [comment]
@bhouston I think it is OK to delete comments you make by mistake. :-)

BTW, it would be interesting to understand the approach by Stephen Hill mentioned in https://github.com/mrdoob/three.js/pull/15277#issuecomment-440081720.

**Comment by bhouston** [comment]
> BTW, it would be interesting to understand the approach by Stephen Hill mentioned in #15277 (comment).

It looks like just a minor curve change.  I think that we can do without, we have worse approximations in Three.JS already. :)  We could make ACES tone mapping the default though going forward.  That would make sense generally.  I believe that in Unreal Engine tone mapping is enabled by default -- although changing defaultsin ThreeJs usually results in a lot of complains about things being broken.  Too bad.

**Comment by mrdoob** [comment]
> @FishOrBear Can you please rebase in order to resolve the merge conflicts?

I'll take care of it.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**